pkgs/applications/misc/comfyui/default.nix

```
@@ -0,0 +1,93 @@
+{ lib
+, python311
+, fetchPypi
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, pkgs
+, torch ? python311.pkgs.torch
+, modelsPath ? "/opt/comfyui/models"
+, inputPath ? "/opt/comfyui/input"
+, outputPath ? "/opt/comfyui/output"
+, customNodesPath ? "/opt/comfyui/custom_nodes"
+}:
+
+let
+  modelPathsFile = writeTextFile {
+    name = "model_paths.yaml";
+    text = (lib.generators.toYAML {} {
+      comfyui = {
+        base_path = modelsPath;
+        checkpoints = "${modelsPath}/checkpoints";
+        clip = "${modelsPath}/clip";
+        clip_vision = "${modelsPath}/clip_vision";
+        configs = "${modelsPath}/configs";
+        controlnet = "${modelsPath}/controlnet";
+        embeddings = "${modelsPath}/embeddings";
+        loras = "${modelsPath}/loras";
+        upscale_models= "${modelsPath}/upscale_models";
+        vae = "${modelsPath}/vae";
+        vae_approx = "${modelsPath}/vae_approx";
+      };
+    });
+  };
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && ${python311}/bin/python comfyui "$@"
+  '';
+in stdenv.mkDerivation rec {
+  pname = "comfyui";
+  version = "8a451234b3090db488fbee9740a5f6be2f989253";
+
+  src = fetchFromGitHub {
+    owner = "comfyanonymous";
+    repo = "ComfyUI";
+    rev = version;
+    hash = "sha256-UuBOPMkbUpfPoQr70OcmakuaGisKiSrXwv4kTuCdzV4=";
+  };
+
+  propagatedBuildInputs = [
+    torch
+  ] ++ (with python311.pkgs; [
+    torchsde
+    torchvision
+    torchaudio
+    transformers
+    safetensors
+    accelerate
+    aiohttp
+    einops
+    pyyaml
+    pillow
+    scipy
+    psutil
+    tqdm
+  ]);
+
+  installPhase = ''
+    echo "Preparing bin folder"
+    mkdir -p $out/bin/
+    echo "Copying comfyui files"
+    cp -r $src/comfy $out/
+    cp -r $src/comfy_extras $out/
+    cp -r $src/web $out/
+    cp -r $src/*.py $out/
+    mv $out/main.py $out/comfyui
+    echo "Copying ${modelPathsFile} to $out"
+    cp ${modelPathsFile} $out/extra_model_paths.yaml
+    echo "Setting up input and output folders"
+    ln -s ${inputPath} $out/input
+    ln -s ${outputPath} $out/output
+    ln -s ${customNodesPath} $out/custom_nodes
+    echo "Copying executable script"
+    cp ${executable}/bin/comfyui $out/bin/comfyui
+    substituteInPlace $out/bin/comfyui --replace "\$out" "$out"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/comfyanonymous/ComfyUI";
+    description = "The most powerful and modular stable diffusion GUI with a graph/nodes interface.";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3;
    platforms = platforms.all;
```
You should probably add yourself as a maintainer, or find someone to maintain this package.